app/controllers/api/LICENSE

```diff
@@ -0,0 +1,19 @@
+Copyright (c) 2012 Brian Ploetz (bploetz@gmail.com)
```

Comment:
@denschub Yeah that sounds clear. Ok so the four folders we need to look through are https://github.com/AugierLe42e/diaspora/tree/openid/app/controllers/api/openid_connect, https://github.com/AugierLe42e/diaspora/tree/openid/app/models/api/openid_connect, https://github.com/AugierLe42e/diaspora/tree/openid/db/migrate, and https://github.com/AugierLe42e/diaspora/tree/openid/lib/api/openid_connect . Nov didn't write any tests so everything 
under spec/ and features/ should be fine. Our front end (most written by @AugierLe42e ) is also fairly different from Nov's, so I think those files should be fine. I'll go file by file so we don't miss anything. Additionally, I'm erring on the side of "maybe" here.

Format: [Attribution needed? Yes/Maybe/No]: [Name of file]: Comment

#### Under [controllers/api/openid_connect](https://github.com/AugierLe42e/diaspora/blob/openid/app/controllers/api/openid_connect/)

- No: [authorizations_controller.rb](https://github.com/AugierLe42e/diaspora/blob/openid/app/controllers/api/openid_connect/authorizations_controller.rb):  This file was inspired by https://github.com/nov/openid_connect_sample/blob/master/app/controllers/authorizations_controller.rb but heavily changed

- No with exception of one method: [clients_controller.rb](https://github.com/AugierLe42e/diaspora/blob/openid/app/controllers/api/openid_connect/clients_controller.rb): Basically one method copied https://github.com/AugierLe42e/diaspora/blob/openid/app/controllers/api/openid_connect/clients_controller.rb#L21 vs https://github.com/nov/openid_connect_sample/blob/master/app/controllers/connect/clients_controller.rb#L24. What should I do here?

- Yes: [discovery_controller.rb](https://github.com/AugierLe42e/diaspora/blob/openid/app/controllers/api/openid_connect/discovery_controller.rb): Should be attributed to https://github.com/nov/openid_connect_sample/blob/master/app/controllers/discovery_controller.rb

- Yes: [id_tokens_controller.rb](https://github.com/AugierLe42e/diaspora/blob/openid/app/controllers/api/openid_connect/id_tokens_controller.rb): Should be attributed to https://github.com/AugierLe42e/diaspora/blob/openid/app/controllers/api/openid_connect/id_tokens_controller.rb)

- No: [token_endpoint_controller.rb](https://github.com/AugierLe42e/diaspora/blob/openid/app/controllers/api/openid_connect/token_endpoint_controller.rb): Basically written by me

- Maybe (Personally No): [user_applications_controller.rb](https://github.com/AugierLe42e/diaspora/blob/openid/app/controllers/api/openid_connect/user_applications_controller.rb): @AugierLe42e It believe this is entirely written by you?

- No: [user_info_controller.rb](https://github.com/AugierLe42e/diaspora/blob/openid/app/controllers/api/openid_connect/user_info_controller.rb): Vaguely inspired by https://github.com/nov/openid_connect_sample/blob/master/app/controllers/user_info_controller.rb. The way we handle scopes and our user/account objects are fairly different.

#### Under [app/models/api/openid_connect](https://github.com/AugierLe42e/diaspora/tree/openid/app/models/api/openid_connect)

- No with exception of one method: [authorization.rb](https://github.com/AugierLe42e/diaspora/blob/openid/app/models/api/openid_connect/authorization.rb): Vaguely inspired by https://github.com/nov/openid_connect_sample/blob/master/app/models/authorization.rb, but that is not why this is a maybe. A similar method from [Nov's access_token.rb](https://github.com/nov/openid_connect_sample/blob/master/app/models/access_token.rb#L26) is used here https://github.com/AugierLe42e/diaspora/blob/openid/app/models/api/openid_connect/authorization.rb#L32. What should I do here?

- Maybe (Personally yes): [id_token.rb](https://github.com/AugierLe42e/diaspora/blob/openid/app/models/api/openid_connect/id_token.rb): Inspired by https://github.com/nov/openid_connect_sample/blob/master/app/models/id_token.rb

- Yes: [o_auth_access_token.rb](https://github.com/AugierLe42e/diaspora/blob/openid/app/models/api/openid_connect/o_auth_access_token.rb): Heavily inspired by https://github.com/nov/openid_connect_sample/blob/master/app/models/access_token.rb

- Maybe (Personally yes): [o_auth_application.rb](https://github.com/AugierLe42e/diaspora/blob/openid/app/models/api/openid_connect/o_auth_application.rb): Inspired by https://github.com/nov/openid_connect_sample/blob/master/app/models/client.rb . What I have done here is added the ability to check the validity of the redirect_uri and sector_identifier_uri, parse and save the jwks_uri, and @AugierLe42e has added in some client attributes.

- Yes: [pairwise_pseudonymous_identifier.rb](https://github.com/AugierLe42e/diaspora/blob/openid/app/models/api/openid_connect/pairwise_pseudonymous_identifier.rb): Heavily inspired by https://github.com/nov/openid_connect_sample/blob/master/app/models/pairwise_pseudonymous_identifier.rb

#### Under [db/migrate](https://github.com/AugierLe42e/diaspora/tree/openid/db/migrate)

I think for these ones, if one of them is yes then all of them are (besides the authorizations migration). I did take a glance at all of Nov's migrations when creating each of these.

- Maybe : [20150731123113_create_pairwise_pseudonymous_identifiers.rb](https://github.com/AugierLe42e/diaspora/blob/openid/db/migrate/20150731123113_create_pairwise_pseudonymous_identifiers.rb): Inspired by https://github.com/nov/openid_connect_sample/blob/master/db/migrate/20110829024140_create_pairwise_pseudonymous_identifiers.rb

- Maybe: [20110829024010_create_id_tokens.rb](https://github.com/nov/openid_connect_sample/blob/master/db/migrate/20110829024010_create_id_tokens.rb): Inspired by https://github.com/nov/openid_connect_sample/blob/master/db/migrate/20110829024010_create_id_tokens.rb

- Maybe: [20150708153928_create_o_auth_access_tokens.rb](https://github.com/AugierLe42e/diaspora/blob/openid/db/migrate/20150708153928_create_o_auth_access_tokens.rb): Inspired by https://github.com/nov/openid_connect_sample/blob/master/db/migrate/20110829023837_create_access_tokens.rb

- No: [20150708153926_create_authorizations.rb](https://github.com/AugierLe42e/diaspora/blob/openid/db/migrate/20150708153926_create_authorizations.rb): Our authorizations model is fairly different from Nov's (Nov's authorization model is essentially a manifestation of the authorization code flow – almost a wrapper for the "secret" code – while ours is "authorization" in a more broad sense). Nov's equivalent migration file would be https://github.com/nov/openid_connect_sample/blob/master/db/migrate/20110829023844_create_authorizations.rb but I wouldn't say they are similar.

- Maybe: [20150613202109_create_o_auth_applications.rb](https://github.com/AugierLe42e/diaspora/blob/openid/db/migrate/20150613202109_create_o_auth_applications.rb): Basically our migration is an extension of Nov's https://github.com/nov/openid_connect_sample/blob/master/db/migrate/20110829023826_create_clients.rb

#### Under [lib/api/openid_connect](https://github.com/AugierLe42e/diaspora/blob/openid/lib/api/openid_connect/)

- No: [authorization_point/endpoint.rb](https://github.com/AugierLe42e/diaspora/blob/openid/lib/api/openid_connect/authorization_point/endpoint.rb): Inspired by https://github.com/nov/openid_connect_sample/blob/master/lib/authorization_endpoint.rb but heavily modified

- No: [authorization_point/endpoint_confirmation_point.rb](https://github.com/AugierLe42e/diaspora/blob/openid/lib/api/openid_connect/authorization_point/endpoint_confirmation_point.rb): Inspired by https://github.com/nov/openid_connect_sample/blob/master/lib/authorization_endpoint.rb but heavily modified

- No: [authorization_point/endpoint_start_point.rb](https://github.com/AugierLe42e/diaspora/blob/openid/lib/api/openid_connect/authorization_point/endpoint_start_point.rb): Inspired by https://github.com/nov/openid_connect_sample/blob/master/lib/authorization_endpoint.rb but heavily modified

- No: [error.rb](https://github.com/AugierLe42e/diaspora/blob/openid/lib/api/openid_connect/error.rb): Basically written by me. No corresponding file in Nov's repo.

- No: [id_token_config.rb](https://github.com/AugierLe42e/diaspora/blob/openid/lib/api/openid_connect/id_token_config.rb): Vaguely inspired by https://github.com/nov/openid_connect_sample/blob/master/app/models/id_token.rb#L74 but fairly different

- Yes: [protected_resource_endpoint.rb](https://github.com/AugierLe42e/diaspora/blob/openid/lib/api/openid_connect/protected_resource_endpoint.rb): This was answered in an earlier comment.

- No: [subject_identifier_creator.rb](https://github.com/AugierLe42e/diaspora/blob/openid/lib/api/openid_connect/subject_identifier_creator.rb): Vaguely inspired by https://github.com/nov/openid_connect_sample/blob/master/app/models/id_token.rb#L18

- Maybe: [token_endpoint.rb](https://github.com/AugierLe42e/diaspora/blob/openid/lib/api/openid_connect/token_endpoint.rb): Inspired by https://github.com/nov/openid_connect_sample/blob/master/lib/token_endpoint.rb but modified

Also how am I supposed to attribute: stick the license file at the top of each file that I need to attribute or put a link the corresponding file at the top? And as I have questioned for some files, what do I do if only one method needs to be attributed?